### PR TITLE
Add SplitButton component

### PR DESCRIPTION
**Pull request by TerrenceLJones**
## Add SplitButton component

This adds a new component called, which is an enhanced version of the V2 Button and V2 Dropdown (with V2 Button trigger). 

### Enabled 
screen recording 2019-02-22 at 10 43 pm

### Sizing and Colour Support
screen shot 2019-02-22 at 10 43 21 pm

### Disabled 

screen recording 2019-02-22 at 10 47 pm

### CODE REVIEW

**Comment by TerrenceLJones** [comment]
@ItsJonQ Not entirely sure about the css on this, would love your expertise!

File: src/components/Dropdown/V2/Dropdown.Trigger.tsx [hunk 1]
```diff
@@ -14,6 +14,7 @@ export interface Props {
   children?: any
   className?: string
   closeDropdown: () => void
+  disabled: boolean
```
Comment (TerrenceLJones) [inline src/components/Dropdown/V2/Dropdown.Trigger.tsx]: @ItsJonQ What are you thoughts on adding this disabled state to the Dropdown component?

**Comment by TerrenceLJones** [comment]
[Add redirect option to End Chat modal](https://trello.com/c/M8ha1Qof/80-add-redirect-option-to-end-chat-modal)

**Comment by ItsJonQ** [comment]
@TerrenceLJones Oo! I like the convenience of this component :). Let me know if you wanna 🍐 ! I didn't leave any comments yet due to the Work in Progress label

**Comment by ItsJonQ** [comment]
@TerrenceLJones Leaving some notes, just in case we don't get to pair today!

I love the concept of this component! 😍 Some suggestions I have are to make it more flexible.

For example, the `Button` UI should be able to render any `kind` and any `size`. `Button`V2 supports a bunch. This will enable us to use this component in many different scenarios. At the moment, it looks like it's locked into `kind="primary"` and `size="lg"`.

The Dropdown should also be able to render a bunch of ways. `Dropdown` (v2) has a ton of simple UI options, like direction, max height/width. I feel like the component should enable to user to adjust these props.

From a functional standpoint, we need to provide the user with a way to hook into when an `option` is selected from the Dropdown via the `onSelect` callback.

More info in the docs!

https://github.com/helpscout/hsds-react/blob/master/src/components/Dropdown/V2/docs/Dropdown.md

**Comment by ItsJonQ** [comment]
@TerrenceLJones Haiii!! Just following up in this one :).
Looks like Travis is failing because the test coverage % has dropped. You can check what you're missing locally by running `npm run test`. Jest will generate a coverage report, which you can view at:

`/Users/Q/Sites/helpscout/hsds-react/coverage/lcov-report/index.html`

It should look like this:

screen shot 2019-02-25 at 8 35 09 am

**Comment by ItsJonQ** [comment]
@TerrenceLJones This is awesome work! Great 🍐 on this one 😍 

One minor thing I'd like to get your thoughts on..

What do you think about renaming the component to `SplitButton`?

I did some Googling, and I believe that's the standard term for this particular UI.

screen shot 2019-02-26 at 9 45 32 am

If it's too much trouble, I can take care of the renaming for you 😊

**Comment by ItsJonQ** [comment]
@TerrenceLJones Ooop! Look slike the Netlify build is failing:

```
2:02:16 PM:  Module not found: Error: Can't resolve '../src/components/ButtonWithOptions' in '/opt/build/repo/stories'
2:02:16 PM: (node:1175) UnhandledPromiseRejectionWarning: ModuleNotFoundError: Module not found: Error: Can't resolve 
```

Something in the stories? :D

**Comment by ItsJonQ** [comment]
cc'ing @digitaldesigner for visual QA/review :D

https://deploy-preview-491--hsds-react.netlify.com/?selectedKind=SplitButton&selectedStory=Default&full=0&addons=1&stories=1&panelRight=0&addonPanel=storybooks%2Fstorybook-addon-knobs

**Comment by digitaldesigner** [comment]
This may be impossible, so forgive my ignorance — but is there any way to prevent the blue border (i.e. the tab focus state) from appearing on click?

**Comment by ItsJonQ** [comment]
@digitaldesigner I'm not sure! It's definitely not conventional.. nor something I've ever thought of doing before!

In other words, you're thinking it should only appear to indicate focus when a user has pressed `Tab`?

Let's say it was possible :).

Would doing so affect accessibility/usability? I think showing the ring helps with that.
I'm open to exploring the idea (prolly on a separate PR)

If you have a codepen, I'd love to check it out! Otherwise, I can come up with a quick prototype

**Comment by digitaldesigner** [comment]
Here you go @ItsJonQ https://www.useloom.com/share/c3b45feb5ca54b858ff89c990dc682af

**Comment by ItsJonQ** [comment]
@digitaldesigner Ohhhhh!!! I gotcha! Not on mousedown.. but on mouseup. Got it! Looking into it now! Thanks!!

(I too was poking at Github's buttons to double check)

**Comment by ItsJonQ** [comment]
Making some updates to `Button` and `Dropdown` to improve focus interactions!

**Comment by ItsJonQ** [comment]
@digitaldesigner 🎉 

screen recording 2019-02-26 at 05 49 pm

**Comment by digitaldesigner** [comment]
My. Man

**Comment by ItsJonQ** [comment]
> My. Man
😂 

With that! I think it's good! Gonna merge it up

**Pull request merged**